spot_rl_experiments/spot_rl/skills/atomic_skills.py

```diff
@@ -264,76 +328,68 @@ def __init__(self, spot, config, use_mobile_pick=False):
 
         self.env = SpotGazeEnv(self.config, spot, use_mobile_pick)
 
-    def reset_env_and_policy(self, target_obj_name):
+    def reset_env_and_policy(self, goal_dict: Dict[str, Any]):
         """
         Resets the env and policy
 
         Args:
-            target_obj_name (str): Name of the target object
+            goal_dict: Dict[str, Any] containing necessary goal information for picking
+                                      Required key: "target_object" (x,y,theta) where robot needs to navigate to
+                                                    "take_user_input" (bool) Whether to take user input for the success of the gaze
 
         Returns:
             observations: observations from the env
 
         """
+        target_obj_name = goal_dict.get("target_object", None)
+        if target_obj_name is None:
+            raise KeyError("Error in Pick.reset_ent_and_policy(): target_obj_name key not found in goal_dict")
+
         observations = self.env.reset(target_obj_name=target_obj_name)
         self.policy.reset()
 
         return observations
 
-    def execute_pick(self, target_object_list, take_user_input=False):
+    def execute_rl_loop(self, goal_dict: Dict[str, Any]):
```

Comment:
Maybe we can define a base skill class in which we have a basic `execute_rl_loop` so that we do not have to define this for every skill we develop. So this saves time and make code a bit more clean?

Reply:
Yes Jimmy, that's the idea.

We were trying to keep the base environment for a future PR so I kept it that way.

We are now in a position to implement base class freely.

Reply:
Added interface class.